pycode/memilio-simulation/memilio/simulation/pybind_util.h

```diff
@@ -52,7 +96,34 @@ decltype(auto) pybind_pickle_class(pybind11::module_& m, const char* name)
                 throw std::runtime_error(object.error().formatted_message());
             }
         }));
-    return pickle_class;
+}
+
+template <class T, class... Args>
+decltype(auto) _bind_class(pybind11::module& m, std::string const& name, NoPicklingTag) {
+    decltype(auto) cls = pybind11::class_<T, Args...>(m, name.c_str());
+    return cls;
+}
+
+template <class T, class... Args>
+decltype(auto) _bind_class(pybind11::module& m, std::string const& name, TryPicklingTag) {
+    decltype(auto) cls = pybind11::class_<T, Args...>(m, name.c_str());
+    // Bind the class depending on its features
+    if constexpr (is_serializable<mio::PickleSerializer, T>::value) {
+        pybind_pickle_class<T, Args...>(cls);
+    }
+    return cls;
+}
+
+template <class T, class... Args>
+decltype(auto) _bind_class(pybind11::module& m, std::string const& name, ForcePicklingTag) {
+    decltype(auto) cls = pybind11::class_<T, Args...>(m, name.c_str());
+    pybind_pickle_class<T, Args...>(cls);
+    return cls;
+}
+
+template <PickleFlag F, class T, class... Args>
+decltype(auto) bind_class(pybind11::module& m, std::string const& name) {
```

Comment:
what's the purpose of decltype(auto) here? Does pybind11::class_ ever return a reference?

Reply:
it was used similar in other parts of the code for pybind11::class_, but i dont think it will ever return a reference.